File: BackEnd/proyecto/src/main/java/com/example/proyecto/modelFB/ChatFB.java
package com.example.proyecto.modelFB;

import lombok.*;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ChatFB {

    private String id;
    private boolean boletin;
    private Map<String, MensajeFB> mensajes;
    private String nombreChat;
    private Map<String, Boolean> usuarios;
    private String foto;

    public Map<String, Object> toMap() {
        Map<String, Object> chatMap = new HashMap<>();
        chatMap.put("id", id); // Incluir id en el mapa
        chatMap.put("boletin", boletin);
        chatMap.put("nombreChat", nombreChat);
        chatMap.put("mensajes", mensajes);
        chatMap.put("usuarios", usuarios);
        chatMap.put("foto", foto); // Incluyendo la variable foto en el mapa
        return chatMap;
    }
}

File: FrontEnd/app/src/main/java/com/example/tycep_fe/modelFB/ChatFB.kt
package com.example.tycep_fe.modelFB

data class ChatFB(
    var id: String? = null, // Cambia el tipo de datos de Int a String
    var boletin: Boolean = false,
    var nombreChat: String = "",
    var mensajes: Map<String, MensajeFB> = emptyMap(),
    var foto: String,
    var usuarios:Map<String, Boolean> = emptyMap()
) {
    // Constructor vacío requerido por Firebase para la deserialización
    constructor() : this("", false, "", emptyMap(), "", emptyMap())


}

